diff --git a/node_modules/optimism/lib/bundle.cjs.js b/node_modules/optimism/lib/bundle.cjs.js
index cb1817e..6aaeb03 100644
--- a/node_modules/optimism/lib/bundle.cjs.js
+++ b/node_modules/optimism/lib/bundle.cjs.js
@@ -513,6 +513,10 @@ function wrap(originalFunction, options) {
             return entry.peek();
         }
     };
+    optimistic.delete = function (key) {
+        const k = makeCacheKey(key);
+        cache.delete(k)
+    }
     return optimistic;
 }
 
diff --git a/node_modules/optimism/lib/bundle.esm.js b/node_modules/optimism/lib/bundle.esm.js
index 99c5962..546fa35 100644
--- a/node_modules/optimism/lib/bundle.esm.js
+++ b/node_modules/optimism/lib/bundle.esm.js
@@ -510,6 +510,10 @@ function wrap(originalFunction, options) {
             return entry.peek();
         }
     };
+    optimistic.delete = function (key) {
+        const k = makeCacheKey(key);
+        cache.delete(k)
+    }
     return optimistic;
 }
 
